### PR TITLE
IMP-2284 Remove EDS and other legacy code from Bento

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

With the launch of Alma and Primo, we no longer need the EDS code
in Bento, and removing it now with save us a major headache in the
future. It also make sense at this point to remove other features
we no longer use, like the button makers, Aleph RTA, etc.

#### Relevant ticket(s):

https://mitlibraries.atlassian.net/browse/IMP-2284

#### How this addresses that need:

This removes several features, including:

* All EDS logic (normalization and search models, as well as code
from the search and record controllers and views, and result model
and views);
* Full records, with the exception of some logic required for Primo
redirects;
* Aleph real-time availability;
* SFX handler;
* Button makers;
* Rainbows; and
* Authentication.

#### Side effects of this change:

* Unused VCR cassettes have been removed.
* The system diagram has been updated.
* Some irrelevant assets have been removed (Aleph RTA JS and SCSS
for removed features).

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: app/views/search/search_boxed.html.erb
Comment (JPrevost) [inline app/views/search/search_boxed.html.erb:54]: Noted elsewhere but I don't think we use TrackLinks anymore. Likely worth a dedicated ticket to come back and clean it up.

File: config/features.rb
Comment (JPrevost) [inline config/features.rb:7]: is debug still used anywhere? I think I saw the partial pulled out but I suppose it may be used elsewhere...

File: config/features.rb
Comment (JPrevost) [inline config/features.rb:10]: I'm curious if both `primo_search` and `primo_redirects` should be removed as features and left as on. I'm happy to hear thoughts on why to keep them though.

**Review by JPrevost — COMMENTED** [review]
Take a look through the comments I've made and let me know if there are any you'd like to address now, open tickets on, or disagree with.

Regardless, before merging let's ask UX to sign off on the functionality in the PR build.

Looks good!

File: app/controllers/record_controller.rb
Comment (jazairi) [reply app/controllers/record_controller.rb:5]: Good point. That seems reasonable to me.

File: app/controllers/search_controller.rb
Comment (jazairi) [reply app/controllers/search_controller.rb:115]: I suspect it is... Will confirm and remove if so.

File: app/models/result.rb
Comment (jazairi) [reply app/models/result.rb:19]: Matt and I discussed it while you were on vacation. The rationale for keeping it is essentially to ease migrations to future discovery backends. I.e., when we migrate from Primo to Catalogtron 4000, we will already have the Primo feature flag in place that we can switch off when we're ready to launch. It's an unlikely scenario that we'll still be using this Bento app at that point, but it seemed relatively harmless to me to leave the feature flag.

This is not a strongly held opinion, and I'm happy to remove the flag if you think it's more confusing than potentially useful. I should note that this and debug mode are the last two feature flags, which prompts the question of whether we should remove Flipflop altogether.

File: app/views/search/_result.html.erb
Comment (jazairi) [reply app/views/search/_result.html.erb:140]: Good catch. I think that also renders the feature flag irrelevant, since we don't use it anywhere else in the app.

File: config/features.rb
Comment (jazairi) [reply config/features.rb:10]: Resolving this one since we have an open discussion in another thread.

File: app/views/search/_result.html.erb
Comment (jazairi) [reply app/views/search/_result.html.erb:140]: Good call, I'll remove that as well as the feature flag.

File: app/views/search/_trigger_search.html.erb
Comment (jazairi) [reply app/views/search/_trigger_search.html.erb:6]: It might be clean enough that it's worth doing in this PR. If not, I'll open a new ticket.

File: app/models/normalize_google.rb
Comment (jazairi) [reply app/models/normalize_google.rb:9]: Yeah. I _think_ that's the last of the hardcoded product names, or at least the last of it that affects anything noticeably...

**Comment by jazairi** [comment]
@JPrevost Thanks for the thorough review! I've addressed most of the comments, but I left the primo_search feature flag until we discuss it further (see my response in the thread above). Still open to removing it -- even more so now that it's the only remaining feature flag -- but wanted to make sure we're considering all angles.

File: app/models/result.rb
Comment (jazairi) [reply app/models/result.rb:19]: Thinking about this further, it might not make a lot of sense to have a feature flag that, when disabled, functionally breaks the app. It's probably more reasonable to introduce a new feature flag for whatever the next catalog is (assuming this app lasts that long).

I would say, though, that along with removing the feature flag, we should probably also remove Flipflop. Does that make sense?

Comment (JPrevost) [reply app/models/result.rb:19]: Yeah let's leave it in for now and reconsider in a few months as we start to get a better feel for how our Discovery Roadmap is looking.

**Review by JPrevost — APPROVED** [review]
This is all good from the code side. Once you get sign off from Darcy/Melissa we can merge. 👍🏻

**Comment by jazairi** [comment]
This now has stakeholder approval, but I have some questions about how we want to handle ENV requirements in app.json moving forward. We'll discuss that in Tuesday's EngX meeting, and this will merge once we have a decision and it's passed final code review.

**Review by JPrevost — APPROVED** [review]

**Pull request merged**
